File: HelpSoftware/Help/Help.Web/Web/src/app/EF/UserSetting.ts
import { Setting, User } from "./EF";

export class UserSetting {
  ID_UserSetting: number;
  Setting: Setting;
  User: User;
  UserValue: string;
  Created: Date;
  LastChanged: Date;

  deserialize(input) {
    this.ID_UserSetting = input.ID_UserSetting;
    this.Setting = new Setting().deserialize(input.Setting);
    this.User = new User().deserialize(input.User);
    this.UserValue = input.UserValue;
    this.Created = input.Created;
    this.LastChanged = input.Date;
  }
}

File: HelpSoftware/Help/Help.EF/Entitys/UserSetting.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Help.EF
{
  public class UserSetting : IHelpEntity
  {
    [Key]
    public virtual int ID_UserSetting { get; set; }

    public virtual Setting Setting { get; set; }

    public virtual User User { get; set; }

    public virtual string UserValue { get; set; }

    public virtual DateTime Created { get; set; }

    public virtual DateTime LastChanged { get; set; }

    [Timestamp]
    public byte[ ] TimeStamp { get; set; }

    public string GetEntityName()
    {
      return "UserSetting";
    }

    public int GetID()
    {
      return ID_UserSetting;
    }

    public bool GetWriteHistoryEntry()
    {
      return true;
    }
  }
}